analysis_engine/library.py

```
@@ -4885,8 +4885,11 @@ def blend_parameters_cubic(frequency, offset, params, result_slice, tolerance=No
         weights.append(blend_parameters_weighting(
             param.array[my_slice], frequency/param.frequency))
 
-    a = np.vstack(tuple(curves))
-    result = np.ma.average(a, axis=0, weights=weights)
+    try:
+        a = np.vstack(tuple(curves))
+        result = np.ma.average(a, axis=0, weights=weights)
+    except:
```

An ideal `try` `except` block should contain a single line in the `try` clause and also catch specific exceptions. This removes ambiguity from the expected behaviour of the code as it is clear which line raises the exception and also which exception is expected to be raised, for instance if someone edited the code above to remove the variable `curves`, a different exception type would be raised rather than the code 'failing silently'. Please change this block to be in the form:
```
try:
    # line which fails
except ExceptionType:
    # exception handling code
else:
    # code to run if the try line did not raise an exception (if an exception is raised here it will propagate)
```